atlasdb-commons/src/main/java/com/palantir/common/remoting/ServiceNotAvailableException.java

```diff
@@ -15,46 +15,78 @@
  */
 package com.palantir.common.remoting;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.net.HostAndPort;
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 
 /**
- * If a server is shutting down or cannot respond to a call for another reson this exception may be thrown.
+ * If a server is shutting down or cannot respond to a call for another reason this exception may be thrown.
  * This exception indicates to the caller that this call should be retried on another server that is available.
  *
  * @author carrino
  */
-public class ServiceNotAvailableException extends RuntimeException {
+public class ServiceNotAvailableException extends RuntimeException implements SafeLoggable {
     private static final long serialVersionUID = 2L;
 
     private final Optional<HostAndPort> serviceHint;
+    private final List<Arg<?>> args;
 
-    public ServiceNotAvailableException(String message, Throwable cause, HostAndPort serviceHint) {
+    public ServiceNotAvailableException(
+            @CompileTimeConstant String message, Throwable cause, HostAndPort serviceHint, Arg<?>... args) {
```

Comment:
This is technically an API break (same for `NCLE`). Could we check on internal code-search tool that we're ok with this?